File: api-openbis-java/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/datastore/create/DataStoreCreation.java
/*
 * Copyright ETH 2008 - 2023 Zürich, Scientific IT Services
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package ch.ethz.sis.openbis.generic.asapi.v3.dto.datastore.create;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.ObjectToString;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.create.ICreation;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.create.IObjectCreation;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author pkupczyk
 */
@JsonObject("as.dto.datastore.create.DataStoreCreation")
public class DataStoreCreation implements ICreation, IObjectCreation
{
    private static final long serialVersionUID = 1L;

    private String code;

    private String downloadUrl;

    private String remoteUrl;

    private String storageUuid;

    public String getCode()
    {
        return code;
    }

    public void setCode(String code)
    {
        this.code = code;
    }

    public String getDownloadUrl()
    {
        return downloadUrl;
    }

    public void setDownloadUrl(final String downloadUrl)
    {
        this.downloadUrl = downloadUrl;
    }

    public String getRemoteUrl()
    {
        return remoteUrl;
    }

    public void setRemoteUrl(final String remoteUrl)
    {
        this.remoteUrl = remoteUrl;
    }

    public String getStorageUuid()
    {
        return storageUuid;
    }

    public void setStorageUuid(final String storageUuid)
    {
        this.storageUuid = storageUuid;
    }

    @Override
    public String toString()
    {
        return new ObjectToString(this).append("code", code).toString();
    }

}

File: api-openbis-javascript/src/v3/as/dto/datastore/create/DataStoreCreation.js
/**
 * @author pkupczyk
 */
define([ "stjs" ], function(stjs) {
	var DataStoreCreation = function() {
	};
	stjs.extend(DataStoreCreation, null, [], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.datastore.create.DataStoreCreation';
		constructor.serialVersionUID = 1;
		prototype.code = null;
		prototype.downloadUrl = null;
		prototype.remoteUrl = null;
		prototype.storageUuid = null;

		prototype.getCode = function() {
			return this.code;
		};
		prototype.setCode = function(code) {
			this.code = code;
		};
		prototype.getDownloadUrl = function() {
			return this.downloadUrl;
		};
		prototype.setDownloadUrl = function(downloadUrl) {
			this.downloadUrl = downloadUrl;
		};
		prototype.getRemoteUrl = function() {
			return this.remoteUrl;
		};
		prototype.setRemoteUrl = function(remoteUrl) {
			this.remoteUrl = remoteUrl;
		};
		prototype.getStorageUuid = function() {
			return this.storageUuid;
		};
		prototype.setStorageUuid = function(storageUuid) {
			this.storageUuid = storageUuid;
		};
	}, {
	});
	return DataStoreCreation;
})